lib/kubeclient.rb

```diff
@@ -52,10 +52,15 @@ def initialize(uri,
       @headers = {}
       @ssl_options = ssl_options
 
-      @basic_auth_user = auth_options[:user]
-      @basic_auth_password = auth_options[:password]
-
-      bearer_token(auth_options[:bearer_token]) if auth_options[:bearer_token]
+      if auth_options[:user]
+        @basic_auth_user = auth_options[:user]
+        @basic_auth_password = auth_options[:password]
+      elsif auth_options[:bearer_token]
+        bearer_token(auth_options[:bearer_token])
+      elsif auth_options[:bearer_token_file] &&
+            File.file(auth_options[:bearer_token_file])
+        bearer_token(File.read(auth_options[:bearer_token_file]))
```

Comment:
> Will add exception handling. Should this fail if files don't exist or just log error & continue?

what is the behavior when you connect to k8s when the token doesn't exist/wrong location? same as passing a bad token/different?